src/views/LotteryV2/hooks/useGetNextLotteryEvent.ts

```diff
@@ -0,0 +1,48 @@
+import { LotteryStatus } from 'config/constants/types'
+import { useTranslation } from 'contexts/Localization'
+import { useEffect, useState } from 'react'
+
+interface LotteryEvent {
+  nextEventTime: number
+  postCountdownText?: string
+  preCountdownText?: string
+}
+
+const useGetNextLotteryEvent = (endTime: number, status: LotteryStatus): LotteryEvent => {
+  const { t } = useTranslation()
+  const vrfRequestTime = 180 // 3 mins
+  const secondsBetweenRounds = 300 // 5 mins
+  const blockBuffer = 3 // Delay countdown by 3s to ensure contract transactions have resolved when countdown finishes
+  const [nextEvent, setNextEvent] = useState({ nextEventTime: null, preCountdownText: null, postCountdownText: null })
+
+  useEffect(() => {
+    // Current lottery is active
+    if (status === LotteryStatus.OPEN) {
+      setNextEvent({
+        nextEventTime: endTime + blockBuffer,
+        preCountdownText: null,
+        postCountdownText: t('until the draw'),
+      })
+    }
+    // Current lottery has finished but not yet claimable
+    if (status === LotteryStatus.CLOSE) {
+      setNextEvent({
+        nextEventTime: endTime + blockBuffer + vrfRequestTime,
+        preCountdownText: t('Winners announced in'),
+        postCountdownText: null,
+      })
+    }
+    // Current lottery claimable. Next lottery has not yet started
+    if (status === LotteryStatus.CLAIMABLE) {
+      setNextEvent({
+        nextEventTime: endTime + blockBuffer + secondsBetweenRounds,
+        preCountdownText: t('Tickets on sale in'),
+        postCountdownText: null,
+      })
+    }
+  }, [status, endTime, t])
```

Comment:
Just checking but no infinite calls here having `t` as a dependency?

Reply:
@hachiojidev `t` is [wrapped](https://github.com/pancakeswap/pancake-frontend/blob/develop/src/contexts/Localization/Provider.tsx#L78) in `useCallback` and therefore should remain it's reference unless you change language

Reply:
Yea nice one both - good query, and I think Cheems is spot on.

The effect fires on language change, but that's it 👍